editor.planx.uk/src/@planx/components/shared/utils.ts

```diff
@@ -63,3 +64,18 @@ export const getPreviouslySubmittedData = ({
 
   return data;
 };
+
+export const characterCounterSwitch = (type: TextInputType | undefined) => {
```

Comment:
Ran round a few circles here trying to get a DRY, maintainable way of filtering out the different `TextInputType` keys, and found this to be the cleanest. 

I could see benefit in adding more complex `TEXT_LIMITS[]` or `props.type` logic at the `<TextInput>` level, but I couldn't get it to work without being cumbersome, or looking like a lot. This felt more readable

Reply:
This is a very fair comment 👍 

This works just fine, but I think the more terse `const showCharacterCount = [TextInputType.ExtraLong, extInputType.Long].includes(type)` is just as clear imho - especially if we were to move this to the `CharacterCount` component.

Alternatively, we could encode this into `TEXT_LIMITS` - 

```ts
export const TEXT_LIMITS = {
  [TextInputType.Short]: {
    showCharacterCount: false,
    limit: 150,
  },
  [TextInputType.Long]: {
    showCharacterCount: true,
    limit: 250,
  },
  [TextInputType.ExtraLong]: {
    showCharacterCount: true,
    limit: 750,
  },
} as const;
```

I think the first (`.includes()`) method would be my preference here, but no strong feelings, this approach works perfectly also 👍